community/kernel/src/test/java/org/neo4j/kernel/impl/core/TestCombinedPropertyTypes.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.core;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.ZoneId;
+
+import org.neo4j.graphdb.Node;
+import org.neo4j.kernel.impl.AbstractNeo4jTestCase;
+import org.neo4j.values.storable.DateTimeValue;
+import org.neo4j.values.storable.DateValue;
+import org.neo4j.values.storable.LocalTimeValue;
+import org.neo4j.values.storable.Value;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestCombinedPropertyTypes extends AbstractNeo4jTestCase
```

Comment:
I really like these tests and it seems like a very useful thing to have for all our types (basically any two different ones in succession). Nothing that this PR needs to have, but it sounds like this behavior has been largely untested.